mechanicalsoup/form.py

```diff
@@ -304,11 +304,19 @@ def choose_submit(self, submit):
             browser.submit_selected()
         """
 
+        submit = (
+            submit or
+            self.form.select_one('input[type="submit"]') or
+            self.form.select_one('button[type="submit"]')
+        )
+
```

Comment:
This is weird, or at least undocumented. The docstring says that `submit` must be a bs4.element.Tag or a string, so there should be no need to look for something else. If the goal is to let the function run when `submit` is not given, then 1) this should be documented, and 2) it would make sense to give a default value to the parameter, like `submit=None`. And then, since the behavior is otherwise to error out when there are multiple matches, I wouldn't do a "`or`" chain, but I'd check that there's one and only one match in `input[type="submit"],button[type="submit"]`. Picking an arbitrary button/submit is very fragile and shouldn't be encouraged.

Reply:
Ah, the doc is here, but in the next commit. Either squash both commits together (probably makes sense), or re-split the commits properly.